### PR TITLE
HPCC-17901 Add ID parameter to DESDL Def widget

### PR DESCRIPTION
**Pull request by miguelvazq**
This will allow the IDE to show specific definition declarations vs. all the defintions and allow the user one less click.

Signed-off by: Miguel Vazquez <miguel.vazquez@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [ ] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browers
  - [ ] The component(s) render as expected

### CODE REVIEW

File: esp/src/eclwatch/DynamicESDLDefinitionQueryWidget.js [hunk 1]
```diff
@@ -107,6 +96,7 @@ define([
                 }
                 context.store.setData(results);
                 context.grid.refresh();
+                context.grid.selectAll();
```
Comment (GordonSmith) [inline esp/src/eclwatch/DynamicESDLDefinitionQueryWidget.js]: @miguelvazq I think you can call ```context.grid.selection([{Id:XXXX}])``` Where "Id" is the Store unique ID?

**Comment by GordonSmith** [comment]
Looks good - not sure if @miguelvazq wants to revisit the "selection" call to only select a single row?

File: esp/src/eclwatch/DynamicESDLDefinitionQueryWidget.js (hunk 1 above)
Comment (miguelvazq) [inline esp/src/eclwatch/DynamicESDLDefinitionQueryWidget.js]: Does the not index the rows?

**Comment by miguelvazq** [comment]
@GordonSmith modified. Using the grid.select() to select the first row. I tried passing in Id as such `context.grid.select({Id:0});` but it fails to select. Whenever using the name (which I am assuming its because the column that is available is "Name") it selects correctly but, I had to query first to insert it dynamically into the select.

File: esp/src/eclwatch/DynamicESDLDefinitionQueryWidget.js [hunk 2]
```diff
@@ -23,11 +23,10 @@ define([
     "hpcc/GridDetailsWidget",
     "hpcc/WsESDLConfig",
     "hpcc/ESPUtil",
-    "hpcc/ECLSourceWidget",
-    
+    "hpcc/DynamicESDLDefinitionDetailsWidget",
```
Comment (GordonSmith) [inline esp/src/eclwatch/DynamicESDLDefinitionQueryWidget.js]: Trailing comma causing build error.

**Comment by GordonSmith** [comment]
@AttilaVamos any reason this PR didn't get a smoke test?

**Pull request merged**